extensions/shodan/src/shodan-api.ts

```diff
@@ -0,0 +1,344 @@
+        throw new Error(`Shodan API error: ${response.status} ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      return data as ShodanScanResponse;
+    } catch (error) {
+      console.error("Error requesting scan:", error);
+      throw new Error(`Failed to request scan: ${error instanceof Error ? error.message : "Unknown error"}`);
+    }
+  }
+
+  async getScanStatus(scanId: string): Promise<ShodanScanStatus> {
+    try {
+      const response = await fetch(`${this.baseUrl}/shodan/scan/${scanId}?key=${this.apiKey}`);
+
+      if (!response.ok) {
+        throw new Error(`Shodan API error: ${response.status} ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      return data as ShodanScanStatus;
+    } catch (error) {
+      console.error("Error getting scan status:", error);
+      throw new Error(`Failed to get scan status: ${error instanceof Error ? error.message : "Unknown error"}`);
+    }
+  }
+
+  async getScanResults(scanId: string, page = 1): Promise<ShodanScanResult> {
+    try {
+      const response = await fetch(
+        `${this.baseUrl}/shodan/host/search?key=${this.apiKey}&query=scan:${scanId}&page=${page}`,
+      );
+
+      if (!response.ok) {
+        throw new Error(`Shodan API error: ${response.status} ${response.statusText}`);
+      }
+
+      const data = (await response.json()) as { matches: ShodanHostInfo[]; total: number };
+      return {
+        matches: data.matches,
+        total: data.total,
+        scan_id: scanId,
+      };
+    } catch (error) {
+      console.error("Error getting scan results:", error);
+      throw new Error(`Failed to get scan results: ${error instanceof Error ? error.message : "Unknown error"}`);
+    }
+  }
+
+  async getAlertInfo(): Promise<ShodanAlertInfo[]> {
+    try {
+  async getAlertInfo(): Promise<ShodanAlertInfo[]> {
+    try {
+      const response = await fetch(`${this.baseUrl}/shodan/alert/info?key=${this.apiKey}`);
+
+      if (!response.ok) {
+        const errorText = await response.text();
+        throw new Error(`Shodan API error: ${response.status} ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      return data as ShodanAlertInfo[];
+    } catch (error) {
+      console.error("Error getting alert info:", error);
+      throw new Error(`Failed to get alert info: ${error instanceof Error ? error.message : "Unknown error"}`);
+    }
+  }
+
+      return data as ShodanAlertInfo[];
+    } catch (error) {
+      console.error("❌ Error getting alert info:", {
+        error: error instanceof Error ? error.message : "Unknown error",
+        stack: error instanceof Error ? error.stack : undefined,
+        timestamp: new Date().toISOString(),
+        errorType: typeof error,
+        errorConstructor: error?.constructor?.name,
+      });
+      throw new Error(`Failed to get alert info: ${error instanceof Error ? error.message : "Unknown error"}`);
+    }
+  }
+}
```

Comment:
remove the extra lines to get the extension to build.
```suggestion
  async getAlertInfo(): Promise<ShodanAlertInfo[]> {
    try {
      const response = await fetch(`${this.baseUrl}/shodan/alert/info?key=${this.apiKey}`);

      if (!response.ok) {
        const errorText = await response.text();
        throw new Error(`Shodan API error: ${response.status} ${response.statusText}`);
      }

      const data = await response.json();
      return data as ShodanAlertInfo[];
    } catch (error) {
      console.error("Error getting alert info:", error);
      throw new Error(`Failed to get alert info: ${error instanceof Error ? error.message : "Unknown error"}`);
    }
  }
}
```